src/ConfiguredMediator.php

```diff
@@ -30,4 +39,22 @@ private function createInstallerFromConfig(Config $config, PackageEvent $event):
             $event
         );
     }
+
+    private function removePhars(): void
+    {
+        $config = Loader::loadFile($this->getMediatorConfig());
+        $binDir = $this->composer->getConfig()->get('bin-dir');
+
+        /** @var \PharIo\ComposerDistributor\File $phar */
+        foreach ($config->phars() as $phar) {
+            $pharLocation = $binDir . DIRECTORY_SEPARATOR . $phar->pharName();
+            if (is_file($pharLocation)) {
+                $this->io->write(sprintf(
+                    'remove phar \'%1$s\'',
+                    $phar->pharName()
+                ));
+                unlink($pharLocation);
```

Comment:
What happens if we don't have write permissions here?

Reply:
good question, I will address this in another pull request ;)

Reply:
We can't unlink. But we were not able to link something there in the first place.... 

Much more interesting is the question whether that value could be modified to remove something "important". But that would mean that the XML-Config-file would need to be tampered with which should only be possible for the creator of the plugin... So: No. Doesn't look like an attack-vector to me....

Reply:
Nobody said anything about attack vector. Just things that can happen: Vou could have run install things as root or any other user, changed ownership or permissions between install and remove, could have a read-only mounts within a container and try scripting remove, ....

Reply:
Oh that wasn't related. the attack vector just came to my mind when I thought about removing files... 😁